### ISSUE TITLE
Support for Readeck

### ISSUE THREAD

**Issue by adamshand** [Fivefold/linkding-injector#26]
Hello,

Not sure if this is within scope, but I've recently started playing with [Readeck](https://readeck.org/en/).  Similar to Linkding, but with a few additional features that are pretty compelling (at least for me).

It has a well [documented and fairly straight forward API](https://codeberg.org/readeck/readeck/src/branch/main/docs/api/bookmarks/routes.yaml).  Is there any interest in adding support?

Thanks,
Adam.

**Comment by Fivefold** [comment]
Hey Adam, some thoughts (tl;dr at the bottom):

Readeck looks like a nice manager with interesting features. Personally, I'm satisfied with the linkding feature set but I might check it out sometime.

My main concerns are two-fold: **feature creep** and - in relation to that - **maintenance effort**. As it is now, I have no automated test framework, which makes deploying new versions quite inefficient as I have to manually test various settings in two browsers and (by now) for a bunch of search engine websites. It's on the top of my list of things to do for the injector, but as I have no experience in writing automated tests for browser extensions, I unfortunately haven't found the time or motivation to dig into the topic.

Getting automated tests would definitely be a requisite for even considering implementing other bookmark services or other major new functionality.

Another aspect is UI/UX. The settings page would need an overhaul to allow choosing the bookmark service for injection and deal with any differences that the services might have.

I'm also not quite sure how high the demand for this would actually be. Gauging from the chrome extension user numbers, Readeck has about 1/10th of the linkding userbase. And the injector extension is used by roughly 1/10th of the linkding users.
That doesn't leave a lot of potential users. While this isn't an important factor for me, I am hesitant to put significant development time into features that I personally don't need and won't be used by most users.

---

tl;dr: I will keep this open as I'm not completely against this but I will also be honest in that you shouldn't expect developments on this any time soon. As always, I'm happy to accept pull requests. If someone wants to use the injector code as a basis for a Readeck fork that's perfectly fine as well. This happened for [miniflux](https://github.com/Sevichecc/miniflux-injector/) as well as [shiori](https://github.com/tezlm/shiori-injector)

**Comment by adamshand** [comment]
Totally understand. 

I've put in a [feature request](https://codeberg.org/readeck/readeck/issues/125) for them to add functionality similar to linkding-injector to their new browser extension.  

If they aren't interested, and I end up moving to Readeck, I can always fork linkding-injector. :-)

**Comment by adamshand** [comment]
In case anyone else is interested and finds this old issue, I've forked linkding-injector to add support for Readeck.    I've just done the first version today, so there will be bugs but wanted to update here so I don't forget to do it later.

https://github.com/adamshand/bookmark-injector
